src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearingpart/AssignHearingPartToSessionAction.java

```diff
@@ -108,36 +109,22 @@ public void act() {
         hearingPart.setSession(targetSession);
         hearingPart.setStatus(statusConfigService.getStatusConfig(Status.Listed));
         hearingPart.setStart(targetSession.getStart());
+        savedHearingPart = hearingPartRepository.save(hearingPart);
     }
 
     @Override //Done although hearing and session for user transactionDAta are not needed
     public List<UserTransactionData> generateUserTransactionData() {
-        HearingPart savedHearingPart = hearingPartRepository.save(hearingPart);
-
-        List<UserTransactionData> userTransactionDataList = new ArrayList<>();
-
-        userTransactionDataList.add(new UserTransactionData("hearing",
-            savedHearingPart.getHearingId(),
-            previousHearing,
-            "lock",
-            "unlock",
-            0)
-        );
-
-        userTransactionDataList.add(new UserTransactionData("hearingPart",
-            savedHearingPart.getId(),
-            previousHearingPart,
-            "update",
-            "update",
-            1)
-        );
+        dataPrepService.prepareLockedEntityTransactionData("hearing", savedHearingPart.getHearingId(), 0);
```

Comment:
Maybe these hearing, hearingPart string could be extracted somewhere as constants.

Reply:
Done